### PR TITLE
Cursor now changes based on what it is touching

### PR DESCRIPTION
**Pull request by NancokPS2**
- The system was changed from using a shape to a point (i can change it to intersect_shape to keep it like it used to be, altho it would make the cursor inconsistent with its actual position)  
- Now uses the physics system directly instead of using an Area2D node
- Currently, the cursor may stop working while the player is dead. (this should be fixed once respawning is implemented)  
  
For testing, hover the mouse cursor over either an NPC, item or enemy. Each should cause the cursor to change.

### CODE REVIEW

**Comment by jonathaneeckhout** [comment]
@NancokPS2  can you rebase and retest now that the respawn is integrated?

**Comment by NancokPS2** [comment]
Can confirm, it still works.

**Comment by NancokPS2** [comment]
Can confirm, it still works.

**Comment by jonathaneeckhout** [comment]
Tested:

issue found:
When loot drops, the cursor is changed to the yellowish cursors. However once the loots has been collected, it stays yellow until you change it like targeting an enemy

**Comment by jonathaneeckhout** [comment]
Screenshot from 2023-10-17 22-17-50

pointer is changed to the interact pointer when milklady is behind the menu

**Comment by NancokPS2** [comment]
Fixed. The first issue was that i assumed items where a JBody2D due to having the "entity_type" property.  
The other issue was solved by always considering the hovered object as null while "JUI.above_ui" is true

File: scripts/classes/JPlayerInput.gd [hunk 1]
```diff
@@ -5,59 +5,120 @@ class_name JPlayerInput
...
+	
+	#Otherwise set it to the appropiate cursor
+	else:
+		match targetCurrent.get("entity_type"):
+			J.ENTITY_TYPE.NPC:
+				cursorToUse = CursorGraphics.TALK
+				hovered_npc.emit(targetCurrent)
+			
+			J.ENTITY_TYPE.ITEM:
+				print_debug("Hovered item")
```
Comment (jonathaneeckhout) [inline scripts/classes/JPlayerInput.gd:72]: Remove this debug line

**Comment by jonathaneeckhout** [comment]
Tested and works fine.

Remove the print_debug and I will merge.

Thanks for your contribution.

**Pull request merged**